src/TrackPanel.cpp

```diff
@@ -1019,8 +1020,9 @@ void TrackPanel::UpdateVRulerSize()
 
 void TrackPanel::OnTrackMenu(Track *t)
 {
+   std::shared_ptr<TrackPanelCell> ptr1(nullptr);
    CellularPanel::DoContextMenu(
-      t ? &ChannelView::Get(*t->GetChannel(0)) : nullptr);
+      t ? ChannelView::Get(*t->GetChannel(0)).GetShared() : ptr1);
```

Comment:
You can simplify this:  you don't need `ptr1` but can just keep `nullptr` in the ternary expression.

Furthermore:  Don't define the function `GetShared` but just use `shared_from_this` directly.

Reply:
As I understand it, the real heart of the bug fix is here:  the shared pointer constructed from the view object extends its lifetime for the duration of the call that pops up the context menu.

Reply:
... So that, inside `DoContextMenu`, `pCellLocal` (or just `pCell`) is still guaranteed to point to a live object.

Reply:
I don't have the code in front of me at the moment but essentially yes.  The crash was occurring across the UI function near the end, because the object that the raw pointer was pointing to disappeared.